src/transformers/optimization.py

```diff
@@ -165,6 +165,53 @@ def lr_lambda(current_step):
     return LambdaLR(optimizer, lr_lambda, last_epoch)
 
 
+def get_polynomial_decay_schedule_with_warmup(
+    optimizer, num_warmup_steps, num_training_steps, lr_end=1e-7, power=2.0, last_epoch=-1
+):
+    """
+    Create a schedule with a learning rate that decreases as a polynomial decay
+    from the initial lr set in the optimizer to end lr defined by `lr_end`,
+    after a warmup period during which it increases linearly from 0 to the
+    initial lr set in the optimizer.
+
+    Args:
+        optimizer (:class:`~torch.optim.Optimizer`):
+            The optimizer for which to schedule the learning rate.
+        num_warmup_steps (:obj:`int`):
+            The number of steps for the warmup phase.
+        num_training_steps (:obj:`int`):
+            The total number of training steps.
+        lr_end (:obj:`float`, `optional`, defaults to 1e-08):
```

Comment:
```suggestion
        lr_end (:obj:`float`, `optional`, defaults to 1e-7):
```
(Does not match the value in the definition.)